app/models/issue_tta_report.rb

```diff
@@ -1,5 +1,54 @@
+require "api_delegator"
+
 class IssueTtaReport < ApplicationRecord
-  validates :issue_id, :tta_report_display_id, :tta_report_id, presence: true
-  validates :tta_report_id, format: {with: /\A[0-9]*\z/, message: "only allows numbers"}
+  validates :issue_id, :tta_report_display_id, presence: true
   validates :tta_report_display_id, uniqueness: {scope: :issue_id}
+  validate :api_call_succeeded
+  validates :tta_report_id, presence: true, if: :errors_blank?
+
+  attr_accessor :access_token
+
+  before_validation :retrieve_tta_report_id
+
+  private
+
+  def api_call_succeeded
+    # do not make the api call here, if @activity_data is nil then we didn't attempt to make a call
+    if !@activity_data.nil?
+      if !activity_data[:success]
+        errors.add(:base, tta_link_error_message(activity_data[:code]))
+      end
+    end
+  end
+
+  def errors_blank?
+    errors.blank?
+  end
+
+  def retrieve_tta_report_id
+    if tta_report_display_id_changed?
+      if activity_data[:success]
+        self.tta_report_id = activity_data[:body][:data][:id]
+      end
+    end
+  end
+
+  def activity_data
+    @activity_data ||= api.request
+  end
+
+  def api
```

Comment:
[thought] should this be `def get`? it is getting the associated activity report, so i would expect that, i think?

Reply:
tuesday morning me doesn't mind `api` as much as friday afternoon me did. up to you!